### PR TITLE
DIG-1322: candig-data-portal updates: MUI, React, NPM, etc

### PR DESCRIPTION
**Pull request by OrdiNeu**
## Ticket(s)

- https://candig.atlassian.net/browse/DIG-1322
- https://candig.atlassian.net/browse/DIG-1424
- https://candig.atlassian.net/browse/DIG-1425

## Description

- This should get rid of most (notably not all) of the security vulnerabilities in candig-data-portal via updating the versions of all of our npm packages. Notably, this update also completes the upgrade from muiv4 to muiv5 by getting rid of the old `useStyles` in favor of the new `styled` components.

## Expected Behaviour

- The website should work precisely the same as it did before, and look the same as well.

## To do/Tickets to be made before merging branch

- Everything more-or-less needs to be checked to make sure it all looks the same.

## Types of Change(s)

-   [x] 🪲 Bug fix (non-breaking change that fixes an issue)
-   [ ] ✨ New feature (non-breaking change that adds functionality)
-   [x] 💥 Breaking change (fix or feature that would cause existing functionality to change)

## Has it been tested for:

-   [ ] My change requires a change to the documentation
-   [ ] I have updated the documentation accordingly
-   [x] Prettier linter doesn't return errors
-   [ ] Production branch PR browser testing: Chrome, Firefox, Edge, etc.
-   [x] Locally tested
-   [ ] Dev server tested
-   [ ] Production tested when merging into stable/production branch

### CODE REVIEW

File: src/views/summary/summary.js
Comment (CourtneyGosselin) [inline src/views/summary/summary.js:1]: It appears the font-weight of chart titles has changed to bold see image from another PR the font-weight is set to regular. This probably changed with all the package changes. image

File: src/views/clinicalGenomic/widgets/sidebar.js
Comment (CourtneyGosselin) [inline src/views/clinicalGenomic/widgets/sidebar.js:1]: This error occurs when you try and use the delete all button from the below image to clear the selections
image

image

File: src/views/clinicalGenomic/widgets/dataVisualization.js
Comment (CourtneyGosselin) [inline src/views/clinicalGenomic/widgets/dataVisualization.js:1]: image
Title font-weight has change to bold it was originally a regular font.

Before:
image

File: src/views/clinicalGenomic/widgets/clinicalData.js
Comment (CourtneyGosselin) [inline src/views/clinicalGenomic/widgets/clinicalData.js:1]: handleRowClick no longer fills in the patient tree on click.

**Review by CourtneyGosselin — CHANGES_REQUESTED** [review]
Found some minor issues when trying out functionality along with some visual difference wit the update

File: src/views/clinicalGenomic/widgets/clinicalData.js
Comment (OrdiNeu) [reply src/views/clinicalGenomic/widgets/clinicalData.js:1]: Fixed in [f2c3ebe](https://github.com/CanDIG/candig-data-portal/pull/111/commits/f2c3ebeff9ffac9a0d0131c1df6e619ad4c07216)

**Review by CourtneyGosselin — APPROVED** [review]
LGTM

**Pull request merged**